src/Features/Core/Portable/Shared/Extensions/DiagnosticAnalyzerExtensions.cs

```diff
@@ -6,57 +6,24 @@ namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
     internal static class DiagnosticAnalyzerExtensions
     {
-        public static DiagnosticAnalyzerCategory GetDiagnosticAnalyzerCategory(this DiagnosticAnalyzer analyzer)
-        {
-            var category = DiagnosticAnalyzerCategory.None;
-
-            if (analyzer is DocumentDiagnosticAnalyzer)
-            {
-                category |= DiagnosticAnalyzerCategory.SyntaxAnalysis | DiagnosticAnalyzerCategory.SemanticDocumentAnalysis;
-            }
-            else if (analyzer is ProjectDiagnosticAnalyzer)
-            {
-                category |= DiagnosticAnalyzerCategory.ProjectAnalysis;
-            }
-            else
-            {
-                if (analyzer is IBuiltInAnalyzer builtInAnalyzer)
-                {
-                    category = builtInAnalyzer.GetAnalyzerCategory();
-                }
-                else
-                {
-                    // It is not possible to know the categorization for a public analyzer,
-                    // so return a worst-case categorization.
-                    category = (DiagnosticAnalyzerCategory.SyntaxAnalysis | DiagnosticAnalyzerCategory.SemanticDocumentAnalysis | DiagnosticAnalyzerCategory.ProjectAnalysis);
-                }
-            }
-
-            return category;
-        }
-
         public static bool SupportsSyntaxDiagnosticAnalysis(this DiagnosticAnalyzer analyzer)
         {
-            var category = analyzer.GetDiagnosticAnalyzerCategory();
-            return (category & DiagnosticAnalyzerCategory.SyntaxAnalysis) != 0;
+            return !(analyzer is ProjectDiagnosticAnalyzer);
         }
 
         public static bool SupportsSemanticDiagnosticAnalysis(this DiagnosticAnalyzer analyzer)
         {
-            var category = analyzer.GetDiagnosticAnalyzerCategory();
-            return (category & (DiagnosticAnalyzerCategory.SemanticSpanAnalysis | DiagnosticAnalyzerCategory.SemanticDocumentAnalysis)) != 0;
+            return !(analyzer is ProjectDiagnosticAnalyzer);
         }
 
         public static bool SupportsSpanBasedSemanticDiagnosticAnalysis(this DiagnosticAnalyzer analyzer)
         {
-            var category = analyzer.GetDiagnosticAnalyzerCategory();
-            return (category & DiagnosticAnalyzerCategory.SemanticSpanAnalysis) != 0;
+            return false;
         }
 
         public static bool SupportsProjectDiagnosticAnalysis(this DiagnosticAnalyzer analyzer)
         {
-            var category = analyzer.GetDiagnosticAnalyzerCategory();
-            return (category & DiagnosticAnalyzerCategory.ProjectAnalysis) != 0;
+            return !(analyzer is DocumentDiagnosticAnalyzer);
```

Comment:
not certain if this is correct.  most things are not DocumentDiagnosticAnalzyers... but they likely should not 'SupportsProjectDiagnosticAnalysis' right?  shoudl this just be 'analyzer is ProjectDiagnosticAnalyzer'?

Reply:
This seems the most concerning bt only because i thought project analyzers were on a significant delay as well.  so would ths cause a large delay for normal language analyzers?

Reply:
This implementation matches the behavior we would get if `analyzer is IBuiltInAnalyzer builtInAnalyzer` always returned false.

Reply:
Ok.  Just wrote out the table and convinced myself you are correct.  :D   So, as long as you odn't see any crazy behavior where this is actually causing 5 second slowdowns, then i'm good.